tests/fail/panic/double_panic.stderr

```diff
@@ -75,7 +75,6 @@ LL |     ABORT();
    = note: inside `std::panicking::rust_panic_with_hook` at RUSTLIB/std/src/panicking.rs:LL:CC
    = note: inside closure at RUSTLIB/std/src/panicking.rs:LL:CC
    = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<[closure@std::rt::begin_panic<&str>::{closure#0}], !>` at RUSTLIB/std/src/sys_common/backtrace.rs:LL:CC
-   = note: inside `std::rt::begin_panic::<&str>` at RUSTLIB/std/src/panicking.rs:LL:CC
```

Comment:
We now prune a but more than before since we remove *all* `#[caller_location]` frames, not just the ones at the top. IMO that makes sense. If it doesn't, we should also adjust the same in CTFE.

Cc @saethlin

Reply:
I agree this makes sense.